abstract_specification.md

```diff
@@ -296,7 +310,7 @@ read(
 Parameters:
 
 - slices - the rows to read. Defaults to 'all'. Rows are addressable with a row offset (uint), a row-offset range (slice), an Arrow array or chunked array of row-offsets, or a list of both offsets and slices.
-- column_names - the named columns to read and return. Defaults to all. The pseudo-column `soma_rowid` may be included in this list.
+- column_names - the named columns to read and return. Defaults to all. System-defined columns (`soma_rowid` and `soma_joinid`) may be included in this list.
```

Comment:
ok sounds like people get back `soma_rowid` and `soma_joinid` on read if they don't specify `column_names` -- correct?

Reply:
Yes - the logic:
* in a `read`, column_names default to "all", ie, all columns
* for SOMADataFrame, both `soma_rowid` and `soma_joinid` are mandatory columns (only the second is materialized).  For SOMAIndexedDataFrame, `soma_joinid` is mandatory.

So by default all columns, including the `soma_` columns are returned.  But the user can exclude them by using a non-default `column_names` param.

Example:

> SOMADataFrame.read()     # returns all use-defined columns plus `soma_rowid` and `soma_joinid`
> SOMADataFrame.read(column_names=['soma_rowid'])     # returns only `soma_rowid`
> SOMAIndexedDataFrame.read(column_names=['soma_joinid'])     # returns only `soma_joinid`
> SOMAIndexedDataFrame.read(column_names=['foo', 'soma_joinid'])     # returns only columsn `foo` and `soma_joinid`
> SOMAIndexedDataFrame.read(column_names=['soma_rowid'])     # generates error as there is no such column

etc.

Reply:
“may be” sounds vague, as if it’s an implementation decision whether the columns are included. Maybe something like “will include […] if they are matched by the user’s criteria”? This could probably be ironed out further but that’s a place to start.

Reply:
how about this as an alternative?
- column_names - the named columns to read and return. Defaults to all, including system-defined columns (`soma_rowid` and `soma_joinid`).

Reply:
@thetorpedodog - sorry, didn't tag you. LMK if the above alternative wording is specific enough.  I agree we need to improve the working, was just trying to wordsmith it a bit more ;-)

Reply:
Your current version sounds good.